mfd_host/feature/cpu/esxi.py

```diff
@@ -62,3 +65,73 @@ def set_numa_affinity(self, numa_state: State) -> None:
             f'esxcli system settings advanced set {value} -o "/Numa/LocalityWeightActionAffinity"',
             custom_exception=CPUFeatureExecutionError,
         )
+
+    def start_cpu_measurement(self) -> RPyCProcess:
+        """
+        Start CPU measurement on SUT host.
+
+        :return: Handle to process
+        """
+        logger.log(level=log_levels.MODULE_DEBUG, msg="Start CPU measurement on SUT Host")
+        return self._connection.start_process(command="esxtop -b -n 8 -d3", log_file=True, shell=True)
+
+    def stop_cpu_measurement(self, process: RPyCProcess, vm_name: str) -> int:
+        """
+        Stop CPU measurement process.
+
+        :param process: Process handle
+        :param vm_name: VM name to filter CPU usage
+        :return: Average CPU usage percentage
+        """
+        logger.log(level=log_levels.MODULE_DEBUG, msg="Stop CPU measurement on SUT Host")
+        if process.running:
+            process.stop()
+            timeout = TimeoutCounter(5)
+            while not timeout:
+                if not process.running:
+                    break
+                sleep(1)
+            else:
+                process.kill()
+                timeout = TimeoutCounter(5)
+                while not timeout:
+                    if not process.running:
+                        break
+                    sleep(1)
+                if process.running:
+                    raise RuntimeError("CPU measurement process is still running after stop and kill.")
+
+        return self.parse_cpu_usage(vm_name=vm_name, process=process)
+
+    def parse_cpu_usage(self, vm_name: str, process: RPyCProcess) -> int:
+        """
+        Parse CPU usage from esxtop output.
+
+        :param vm_name: VM name to filter CPU usage
+        :param process: Process handle
+        :return: average CPU usage percentage
+        """
+        parsed_file_path = "/tmp/parsed_output.txt"
+        command = (
+            f"cut -d, -f`awk -F, '{{for (i=1;i<=NF;i++){{if ($i ~/Group Cpu.*{vm_name}).*Used/) "
+            f"{{print i}}}}}}' {process.log_path}` {process.log_path}>{parsed_file_path}"
+        )
+        self._connection.execute_command(command=command, shell=True)
+        p = self._connection.path(process.log_path)
+        p.unlink()
+        try:
+            with self._connection.modules().builtins.open(parsed_file_path, "r") as f:
```

Comment:
I recommend to use path API for reading the output, to allow using other connection than RPyC.
```suggestion
            parsed_file_path = self._connection.path(parsed_file_path)
            file_content = parsed_file_path.read_text()
```

Reply:
done